test/cctest/wasm/test-run-wasm.cc

```diff
@@ -338,6 +272,137 @@ class WasmFunctionCompiler : public HandleAndZoneScope,
 };
 
 
+// A helper class to build graphs from Wasm bytecode, generate machine
+// code, and run that code.
+template <typename ReturnType>
+class WasmRunner {
+ public:
+  WasmRunner(MachineType p0 = kMachNone, MachineType p1 = kMachNone,
+             MachineType p2 = kMachNone, MachineType p3 = kMachNone) :
+    signature_(MachineTypeForC<ReturnType>() == kMachNone ? 0 : 1,
+               GetParameterCount(p0, p1, p2, p3),
+               storage),
+    compiler_(&signature_),
+    call_wrapper_(p0, p1, p2, p3) {
+
+    int index = 0;
+    MachineType ret = MachineTypeForC<ReturnType>();
+    if (ret != kMachNone) {
+      storage[index++] = LocalTypeFor(ret);
+    }
+    if (p0 != kMachNone) storage[index++] = LocalTypeFor(p0);
+    if (p1 != kMachNone) storage[index++] = LocalTypeFor(p1);
+    if (p2 != kMachNone) storage[index++] = LocalTypeFor(p2);
+    if (p3 != kMachNone) storage[index++] = LocalTypeFor(p3);
+  }
+
+
+  FunctionEnv* environment() { return &compiler_.env; }
+
+
+  // Builds a graph from the given Wasm code, and generates the machine
+  // code and call wrapper for that graph. This method must not be called
+  // more than once.
+  void Build(const byte* start, const byte* end) {
```

Comment:
Can you either enforce the restriction of a single call (with a DCHECK) or make it cache the result so that you can?